feat(i18n): I18n used only if that module is installed and initialized

doktordirk opened the pull request:
not 100% sure if new Optional(I18N) is the best way, but i didn't spot an example which used that feature.

doktordirk committed:
feat(i18n): I18n used only if that module is installed and initialized

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,27 @@
   }
 
+  _log(message, options) {
+    if (this.i18n && this.i18n.i18next.isInitialized())
```
Curly braces

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,27 @@
     this.i18n   = i18n;
   }
 
+  _log(message, options) {
```
Remove the underscore

RWOverdijk commented:
If we want to do this, we have to also [remove i18n as a dependency](https://github.com/SpoonX/aurelia-notification/blob/master/package.json#L54).

doktordirk commented:
right.

RWOverdijk commented:
@doktordirk Also, thank you. :D

doktordirk committed:
chore(index): Linted

doktordirk committed:
doc(README): Update i18n note

doktordirk commented:
re. i18n dependency. as it is it has to stay. my code still has/needs `import {I18N} from 'aurelia-i18n';`. Wouldn't work without some aurelia-i18n somewhere.

doktordirk committed:
chore(project): Gulp tasks and packages updated

doktordirk committed:
fix(project): Use system module format

doktordirk committed:
chore(project): New build

doktordirk committed:
chore(project): Bump version

RWOverdijk commented on doc/CHANGELOG.md:
```diff
@@ -0,0 +1,14 @@
+<a name"0.0.2"></a>
```
? Why is this here?

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,28 @@
 export class Notification {
-  constructor (humane, i18n) {
-    this.humane = humane;
+  constructor(_humane, i18n) {
```
Why do you like underscores so much? :p Please take out the underscore :)

RWOverdijk replied:
That's fair; import as Humane.

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,28 @@
 import humane from 'wavded/humane-js';
 import {inject} from 'aurelia-framework';
 import {I18N} from 'aurelia-i18n';
+import {Optional} from 'aurelia-dependency-injection';
```
Remove this

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,28 @@
+import {Optional} from 'aurelia-dependency-injection';
 
-@inject(humane, I18N)
+@inject(humane, new Optional(I18N))
```
Remove new Optional(

RWOverdijk commented on src/index.js:
```diff
@@ -1,19 +1,28 @@
-  success (message) {
-    this.humane.log(this.i18n.tr(message), {addnCls: 'success'});
+  log(message, options) {
+    if (this.i18n && this.i18n.i18next.isInitialized()) {
```
Remove `this.i18n &&`; but keep `this.i18n.i18next.isInitialized()`. That way, we only use it if it's configured and it automatically works.

doktordirk replied:
doesn't work. Maybe it's just a bug in aurelia-i18n but as of yesterday it crashes on the new i18n() in inject(I18n) if i18n wasn't setup in main.js  with at least .plugin('aurelia-i18n', () => {})

doktordirk commented:
besides the old gulp lint complaining about humane being previously defined, i also dont like to use humane for the class and the instance. hmm, but i could try import humane as Humane or so. but then, got a new version already which might need underscores. 
as for <a name"0.0.2"></a>. those are added by conventional-changelog >= 0.0.13. , doesn't seem to do much harm
re gitter: i'll quickly add my (not too much changed last version in case it'll help)

doktordirk committed:
feat(notification): Allow Array of Strings

doktordirk commented:
about injecting i18n maybe see https://github.com/aurelia/i18n/issues/48#issuecomment-169722881

doktordirk committed:
fix(project): Re-add lost i18n dependency

doktordirk committed:
chore(index): Linted

doktordirk committed:
doc(README): Added information about styles

doktordirk committed:
doc(notification): Added api documentation

doktordirk committed:
chore(doc): Moved to right  folder

doktordirk committed:
feat(config): Added plugin configuration

doktordirk commented:
other than the unresolved i18n bug causing the Optional(I18n) i better be finished for now

doktordirk committed:
chore(project): Dependency update

doktordirk committed:
refactor(i18n): Removed Optional

aurelia-i18n 0.4.6 will have the issue of uninitialzed i18n fixed
(https://github.com/aurelia/i18n/commit/466e6ae5ce974f023f94a43dacd89c999e9b0959)
